marin/speedrun/slice_cache.py

```diff
@@ -0,0 +1,232 @@
+"""
```

Comment:
Ca you say more about how slicing works...what is the input and what's the output and guarantees?